docs/modules/ROOT/pages/how-tos/opsgenie.adoc

```diff
@@ -0,0 +1,263 @@
+
+[source,yaml]
+----
+openshift4_monitoring:
+  alertManagerConfig:
+    route:
+      group_by:
+        - alertname
+        - namespace
+        - severity
+----
+
+First we want to map the alert's severity label to an OpsGenie priority.
+OpsGenie priorities are `P1`, `P2`, `P3` and `P4` in descending order of urgency.
+We want to map `critical` severity to `P1`, `warning` to `P2`, `info` to `P3` and everything else (this includes alerts which don't have a severity label) to `P4`.
+To achieve this mapping we add the following configuration in the opsgenie receiver:
+
+[source,yaml]
+----
+openshift4_monitoring:
+  alertManagerConfig:
+    receivers:
+      - name: opsgenie
+        opsgenie_configs:
+          - priority: '{{ if eq .GroupLabels.severity "critical" }}P1{{ else if eq .GroupLabels.severity "warning" }}P2{{ else if eq .GroupLabels.severity "info" }}P3{{ else }}P4{{ end }}'
+----
+
+Next, we want to have a title for the OpsGenie alerts which gives some Project Syn information at a glance (tenant and cluster):
+
+[source,yaml]
+----
+openshift4_monitoring:
+  alertManagerConfig:
+    receivers:
+      - name: opsgenie
+        opsgenie_configs:
+          - message: '[{{ .CommonLabels.tenant_id }}/{{ .CommonLabels.cluster id }}] {{ .GroupLabels.alertname }} in {{ .GroupLabels.namespace }}'
+----
+
+Because the https://github.com/prometheus/alertmanager/blob/master/template/default.tmpl[default Alertmanager template] for OpsGenie alert descriptions doesn't fully match our use case, we deploy a custom template for the alert description.
+
+[source,yaml]
+----
+openshift4_monitoring:
+  alertManagerConfig:
+    receivers:
+      - name: opsgenie
+        opsgenie_configs:
+          - description: |-
+              {{ if gt (len .Alerts.Firing) 0 -}}
+              Alerts Firing:
+              {{ range .Alerts.Firing }}
+               - Message: {{ .Annotations.message }}
+                 Labels:
+              {{ range .Labels.SortedPairs }}   - {{ .Name }} = {{ .Value }}
+              {{ end }}   Annotations:
+              {{ range .Annotations.SortedPairs }}   - {{ .Name }} = {{ .Value }}
+              {{ end }}   Source: {{ .GeneratorURL }}
+              {{ end }}
+              {{- end }}
+              {{ if gt (len .Alerts.Resolved) 0 -}}
+              Alerts Resolved:
+              {{ range .Alerts.Resolved }}
+               - Message: {{ .Annotations.message }}
+                 Labels:
+              {{ range .Labels.SortedPairs }}   - {{ .Name }} = {{ .Value }}
+              {{ end }}   Annotations:
+              {{ range .Annotations.SortedPairs }}   - {{ .Name }} = {{ .Value }}
+              {{ end }}   Source: {{ .GeneratorURL }}
+              {{ end }}
+              {{- end }}
+----
+
+To make alerts filterable, we add a number of key-value pairs as `details` and a number of values as `tags`.
```

Comment:
Can you add a reference to the change @srueg mentioned? The one that automatically adds all tags as details.

Reply:
https://github.com/prometheus/alertmanager/pull/2276